### PR TITLE
add exe_path to get executable path from downloaded or provided by env

### LINKED ISSUE #40: Add `exe_path` method
As something like

```
fn exe_path() -> Option<String> {
        if let Some(downloaded_exe_path) = bitcoind::downloaded_exe_path() {
            downloaded_exe_path
        } else {
            env::var("BITCOIND_EXE").expect(
                "when no version feature is specified, you must specify BITCOIND_EXE env var",
            )
        }
    }
```

### PR DESCRIPTION
**Pull request by RCasatta**
close #40

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 name: Test
 
-on: [push, pull_request]
+on: [pull_request]
```
Comment (LeoComandini) [inline .github/workflows/test.yml:3]: Is this intended? if so maybe split it in another commit

Comment (RCasatta) [reply .github/workflows/test.yml:3]: yes, I noticed it cause to start the CI 2 times in most case, will split commits

File: src/lib.rs [hunk 2]
```diff
@@ -314,14 +314,19 @@ pub fn downloaded_exe_path() -> Option<String> {
     }
 }
 
+/// Returns the daemons executable path, if it's provided as a feature or as `BITCOIND_EXE` env var
+/// If both are set, the one provided by the feature is returned
```
Comment (LeoComandini) [inline src/lib.rs:318]: nit
```suggestion
/// Returns the daemon executable path. If both `BITCOIND_EXE` env var and a feature are set, 
/// the path provided by the feature is returned.
```

**Comment by LeoComandini** [comment]
utACK f25801a941b43447840698b1a6a091462c91f46d with nits
